### ISSUE TITLE
Wrong region is used by MinioClient

### ISSUE THREAD

**Issue by tobias-** [minio/minio-java#670]
We get `The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'eu-west-1'` messages. 

By adding debug statements, the bug can be confirmed to be in the minio code, as we get the correct data from amazon. The method updateRegionCache() does not give the correct result though. It's kind of hard to reproduce, but was 100% reproducible in our pre-production, but impossible to reproduce locally. Adding debug statements affects the bug.

Fixed in #669.

Okhttp communication:
```
--> GET https://s3.amazonaws.com/*REDACTED*/?location= http/1.1 
Host: s3.amazonaws.com 
User-Agent: Minio (amd64; amd64) minio-java/dev 
x-amz-content-sha256: UNSIGNED-PAYLOAD 
x-amz-date: 20180404T182842Z 
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20180404/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED* 
Connection: Keep-Alive 
Accept-Encoding: gzip 
--> END GET 
<-- 200 OK https://s3.amazonaws.com/*REDACTED*/?location= (77ms) 
x-amz-id-2: *REDACTED* 
x-amz-request-id: 966A7AC968FA8BFE 
Date: Wed, 04 Apr 2018 18:28:43 GMT 
Content-Type: application/xml 
Transfer-Encoding: chunked 
Server: AmazonS3 
OkHttp-Sent-Millis: 1522866522661 
OkHttp-Received-Millis: 1522866522739 
 
<?xml version="1.0" encoding="UTF-8"?> 
<LocationConstraint xmlns="http://s3.amazonaws.com/doc/2006-03-01/">eu-west-1</LocationConstraint> 
<-- END HTTP (137-byte body) 




--> PUT https://*REDACTED*.s3.amazonaws.com/*REDACTED* http/1.1 
Content-Type: application/octet-stream 
Content-Length: 21183 
Content-Encoding: aws-chunked 
x-amz-decoded-content-length: 21008 
Host: *REDACTED*.s3.amazonaws.com 
User-Agent: Minio (amd64; amd64) minio-java/dev 
x-amz-content-sha256: STREAMING-AWS4-HMAC-SHA256-PAYLOAD 
x-amz-date: 20180404T182842Z 
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20180404/us-east-1/s3/aws4_request, SignedHeaders=content-encoding;host;x-amz-content-sha256;x-amz-date;x-amz-decoded-content-length, Signature=*REDACTED* 
Connection: Keep-Alive 
Accept-Encoding: gzip 
--> END PUT (encoded body omitted) 
<-- 400 Bad Request https://*REDACTED*.s3.amazonaws.com/*REDACTED* (8ms) 
x-amz-request-id: DEC7F643754C4ABE 
x-amz-id-2: *REDACTED* 
Content-Type: application/xml 
Transfer-Encoding: chunked 
Date: Wed, 04 Apr 2018 18:28:41 GMT 
Connection: close 
Server: AmazonS3 
OkHttp-Sent-Millis: 1522866522844 
OkHttp-Received-Millis: 1522866522852 
 
<?xml version="1.0" encoding="UTF-8"?> 
<Error><Code>AuthorizationHeaderMalformed</Code><Message>The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'eu-west-1'</Message><Region>eu-west-1</Region><RequestId>DEC7F643754C4ABE</RequestId><HostId>hmRz1poELa9+g80BVbU2QqErse7tEqEF1F7+1WZZs/UJTgoRAjIDR0K0z7xifrv1GqroxQJ1X8g=</HostId></Error> 
<-- END HTTP (365-byte body) 
```

**Comment by balamurugana** [comment]
```xml
<?xml version="1.0" encoding="UTF-8"?> <Error><Code>AuthorizationHeaderMalformed</Code><Message>The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'eu-west-1'</Message><Region>eu-west-1</Region><RequestId>DEC7F643754C4ABE</RequestId><HostId>hmRz1poELa9+g80BVbU2QqErse7tEqEF1F7+1WZZs/UJTgoRAjIDR0K0z7xifrv1GqroxQJ1X8g=</HostId></Error>
```

It is strange to see this error.  There could be two reason where an bucket location cache went wrong.

1. (updateRegionCache())[https://github.com/minio/minio-java/blob/master/api/src/main/java/io/minio/MinioClient.java#L1211] failed to parse returned XML.  May be due to the issue mentioned and fixed in https://github.com/minio/minio-java/pull/669.
2. A bucket is removed and recreated in other than `us-east-1` reason in between.

For `1`, does PR https://github.com/minio/minio-java/pull/669 fix this issue as well?

Could you confirm whether option `2` happened?

**Comment by deekoder** [comment]
ping @tobias- can you confirm if 1 or 2 happened above?

**Comment by deekoder** [comment]
ok got it @tobias- so we have merged #669 and made an official minio-java sdk release. Hence will close as fixed.
